nengo_loihi/hardware/simulator.py

```diff
@@ -0,0 +1,441 @@
+from __future__ import division
+
+from distutils.version import LooseVersion
+import logging
+import os
+import sys
+import time
+import warnings
+
+import jinja2
+import numpy as np
+
+from nengo.exceptions import SimulationError
+
+try:
+    import nxsdk
+    from nxsdk.arch.n2a.graph.probes import N2SpikeProbe
+
+except ImportError:
+    exc_info = sys.exc_info()
+
+    def no_nxsdk(*args, **kwargs):
+        raise exc_info[1]
+    nxsdk = N2Board = BasicSpikeGenerator = TraceCfgGen = N2SpikeProbe = (
+        no_nxsdk)
+
+from nengo_loihi.compartments import CxGroup
+from nengo_loihi.discretize import scale_pes_errors
+from nengo_loihi.hardware.allocators import one_to_one_allocator
+from nengo_loihi.hardware.builder import build_board
+from nengo_loihi.probes import CxProbe
+
+logger = logging.getLogger(__name__)
+
+
+class LoihiSimulator(object):
```

Comment:
Occurred to me that since we're separating these by submodule now, we could just have `hardware.Simulator` and `emulator.Simulator`.  Pro is that it makes the relation between all these different simulators more clear, but con is that if you import the simulator directly (e.g. `from emulator import Simulator`) then it isn't as obvious in the code what type of Simulator you're working with.  I'm on the fence, don't feel strongly either way.  But thought I'd raise the possibility in case anyone else has opinions.

Reply:
I'd vote to leave it as-is. When we use these, we often import both, so we'd need to rename things anyway or use full names.

Reply:
`EmulatorInterface` and `HardwareInterface`